tooling/cli/src/icon.rs

```diff
@@ -154,8 +162,16 @@ fn ico(source: &DynamicImage, out_dir: &Path) -> Result<()> {
 
 // Generate .png files in 32x32, 128x128, 256x256, 512x512 (icon.png)
 // Main target: Linux
-fn png(source: &DynamicImage, out_dir: &Path) -> Result<()> {
-  for size in [32, 128, 256, 512] {
+fn png(source: &DynamicImage, out_dir: &Path, mut extra_icon_sizes: Vec<u32>) -> Result<()> {
+  //remove duplicate extra icons
+  let mut sizes = vec![32, 16, 24, 48, 64, 256];
+  extra_icon_sizes = extra_icon_sizes
+    .into_iter()
+    .filter(|x| !sizes.contains(x))
+    .collect::<Vec<u32>>();
+  sizes.append(extra_icon_sizes.as_mut());
```

Comment:
it may be nicer to do something like this
```rs
  sizes.append(extra_icon_sizes.as_mut());
  sizes.sort_unstable();
  sizes.dedup();
```
what do you think?

Reply:
- I think .filter() takes linear time. Idk how much it cost to cast from vec to iterator and back. 
- Sorting may take longer, and extra cost on dedup()

Reply:
i don't think performance on a 5 entry vec matters, it's only about the esthetics :D

really doesn't matter, it was just a random thought.

Reply:
Users may input tons of extra sizes, but that may not happen :D . It's just my habit for best practice.